### PR TITLE
Rename `ParallelPlugin` to `ParallelStrategy`

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Follow up to #11120
Part of #10549

Goes last with all changelog updates

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

Part of #1 (it's a lie, this is just here to avoid noisy GitHub bot)

cc @justusschock @awaelchli @akihironitta @borda

### CODE REVIEW

**Review by ananthsub — COMMENTED** [review]
On naming, I think we have inconsistencies across components in Lightning

Loggers, Profilers, and ClusterEnvironments are suffixed with the component name (e.g. TensorboardLogger, SimpleProfiler, SlurmEnvironment)

Here it seems like all strategies will no longer have the suffix

- Would we also make this change for other components of Lightning so the naming scheme is unified across components (leading to one less thing to think about for users) ? 
- Do you think there's a risk with naming collisions? For instance, the IPU accelerator and IPU strategy could both be IPU classes. As a result, users would likely need to change the import for them to be:

```
from pytorch_lightning.accelerators import IPU as IPUAccelerator
from pytorch_lightning.strategy import IPU as IPUStrategy
```

**Comment by kaushikb11** [comment]
Shouldn't we have it as `ParallelStrategy`?

**Review by carmocca — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Review by ananthsub — APPROVED** [review]

**Review by akihironitta — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -136,7 +136,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 
 - Renamed the `TrainingTypePlugin` to `Strategy` ([#11120](https://github.com/PyTorchLightning/pytorch-lightning/pull/11120))
-
+    * Renamed `ParallelPlugin` to `ParallelStrategy` ([#11123](https://github.com/PyTorchLightning/pytorch-lightning/pull/11123))
```
Comment (tchaton) [inline CHANGELOG.md:139]:
```suggestion
- Renamed `ParallelPlugin` to `ParallelStrategy` ([#11123](https://github.com/PyTorchLightning/pytorch-lightning/pull/11123))
```

Comment (tchaton) [reply CHANGELOG.md:139]: Let's group all Accelerator Refactor Entries together in the CHANGELOG, this would get confusing for the readers.

**Review by tchaton — APPROVED** [review]
LGTM !

File: CHANGELOG.md (hunk 1 above)
Comment (awaelchli) [reply CHANGELOG.md:139]: You are suggesting not to group them?

Comment (awaelchli) [reply CHANGELOG.md]: Grouped them together now. Some PRs have been merged out of order

**Pull request merged**